tests/PHPStan/Rules/DeadCode/UnreachableStatementRuleTest.php

```diff
@@ -149,6 +149,34 @@ public function testBug8620(): void
 		$this->analyse([__DIR__ . '/data/bug-8620.php'], []);
 	}
 
+	public function testBug4002(): void
+	{
+		$this->treatPhpDocTypesAsCertain = true;
+		$this->analyse([__DIR__ . '/data/bug-4002.php'], []);
+		$this->analyse([__DIR__ . '/data/bug-4002-2.php'], []);
+		$this->analyse([__DIR__ . '/data/bug-4002-3.php'], [
+			[
+				'Unreachable statement - code above always terminates.',
+				13,
+			],
+		]);
+		$this->analyse([__DIR__ . '/data/bug-4002-4.php'], [
+			[
+				'Unreachable statement - code above always terminates.',
+				9,
+			],
+		]);
+		$this->analyse([__DIR__ . '/data/bug-4002_class.php'], []);
+		$this->analyse([__DIR__ . '/data/bug-4002_interface.php'], []);
+		$this->analyse([__DIR__ . '/data/bug-4002_trait.php'], []);
```

Comment:
For each `test*` method, only a single `analyse()` call should be made. So these calls should each have their own test method.

Reply:
I've split the method.